tests/PPC/TestLifting.cpp

```diff
@@ -0,0 +1,800 @@
+         }},
+        {"cr7",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr7;
+         }},
+        {"lr",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.lr.qword;
+         }},
+        {"ctr",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.ctr.qword;
+         }},
+        {"xer",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.xer.qword;
+         }},
+        {"xer_so",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.xer_flags.so;
+         }},
+        {"xer_ov",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.xer_flags.ov;
+         }},
+};
+
+std::optional<remill::Instruction>
+GetFlows(std::string_view bytes, uint64_t address, uint64_t vle_val) {
+
+  llvm::LLVMContext context;
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  context.enableOpaquePointers();
+#endif
+  auto arch = remill::Arch::Build(&context, remill::OSName::kOSLinux,
+                                  remill::ArchName::kArchPPC);
+  auto sems = remill::LoadArchSemantics(arch.get());
+
+
+  remill::DecodingContext dec_context;
+  dec_context.UpdateContextReg(std::string("VLEReg"), vle_val);
+  CHECK(dec_context.HasValueForReg("VLEReg"));
+  remill::Instruction insn;
+
+  if (!arch->DecodeInstruction(address, bytes, insn, dec_context)) {
+    return std::nullopt;
+  } else {
+    return insn;
+  }
+}
+}  // namespace
+
+using test_runner::TestOutputSpec;
+
+template <typename State, typename = typename std::enable_if_t<
+                              std::is_base_of<ArchState, State>::value>>
+class TestSpecRunner {
+ private:
+  test_runner::LiftingTester lifter;
+  uint64_t tst_ctr;
+  test_runner::random_bytes_engine rbe;
+  llvm::support::endianness endian;
+
+ public:
+  TestSpecRunner(llvm::LLVMContext &context, remill::ArchName archname)
+      : lifter(test_runner::LiftingTester(context, remill::OSName::kOSLinux,
+                                          archname)),
+        tst_ctr(0),
+        endian(lifter.GetArch()->MemoryAccessIsLittleEndian()
+                   ? llvm::support::endianness::little
+                   : llvm::support::endianness::big) {}
+
+  void RunTestSpec(const TestOutputSpec<State> &test) {
+    std::stringstream ss;
+    ss << "test_disas_func_" << this->tst_ctr++;
+
+    auto maybe_func =
+        lifter.LiftInstructionFunction(ss.str(), test.target_bytes, test.addr);
+
+
+    CHECK(maybe_func.has_value());
+    auto lifted_func = maybe_func->first;
+
+    auto new_mod = llvm::CloneModule(*lifted_func->getParent());
+    remill::OptimizeBareModule(new_mod.get());
+
+    auto justFuncMod =
+        std::make_unique<llvm::Module>("", new_mod->getContext());
+
+    auto new_func = test_runner::CopyFunctionIntoNewModule(
+        justFuncMod.get(), lifted_func, new_mod);
```

Comment:
On another note, I realise this section is copied from elsewhere, but it could use some comments. I find this bit where it's copying the lifted function to a new module a bit puzzling.